### DISCUSSION TITLE
Image load inconsistent

### DISCUSSION THREAD

**Discussion by marinkreso95** [zauberzeug/nicegui#3334] [Q&A]
### Question

Hi I have issue with inconsistent image loading (ui.image) of deployed NiceGUI app on Azure:

Context:

- I deployed NiceGUI app on Azure
- I notice some images won't load (I know they exist) on first try, blank space instead is shown
- If I try again, sometimes they show, but often not.
- Locally everything works perfectly
- I know for sure image exists. 
- Image is served from the local storage (included with deployment). Also sometimes it will load as I said, so I'm sure it exists
- Images are not loaded through http/blob_stroage, so I guess there is no latency issues regarding image fetch.
- I'm using tabs and loading images within tabs, might be important

1. Is it problem with Azure and rate limiting or something? 
2. Does anybody else have similar problems? 
3. My app has tabs and maybe a lot of images in total (around 20-30), is maybe this the problem? I load many images at once (although is 20-30 images many or not)

Not sure if problem is NiceGUI or Azure. Thank you for the help!

Example:

```py
with ui.tab_panel('tab1'):
  ui.image(image_one)
  ui.text('lorem ipsum')
with ui.tab_panel('tab2'):
  ui.image(image_two)
  ui.text('lorem ipsum another')
 ```

**Comment by rodja** [comment]
Do you run with multiple instances or workers in the Azure cloud? That could make the behaviour so erratic.

**Reply by marinkreso95** [reply]
Thank you @rodja for a response. I changed the logic to use Azure Blob storage instead and everything works well. 

To answer your question, yes, I was using multiple workers. I'll do a test with single worker and get back to you.

**Comment by BKLronin** [comment]
I have the same behaviour on  a local docker served site. https://app.raise-uav.com

I thought I had this fixed with a python update inside the container but just discovered it still persists.

**Reply by rodja** [reply]
Verify that you have only a single worker and not multiple instances running.

**Reply by BKLronin** [reply]
Thanks, I have it running like so at the moment: `CMD ["gunicorn", "--workers=1", "--bind=0.0.0.0:5000", "main:app"]` In the Dockerfile.

People testing report as all pictures showing but when inspecting with dev tools in vivaldi (chrome) I get 404s but only when scrolling. I can forcefully reload each pages picture with F5. 

I was worried a new visitor would experience the same like in disabled cache mode.

Full Folder is added like so `app.add_static_files('/static', 'static')` but I also have some elements added in like this to achive background functionality and get a path : 

```py
path = app.add_media_file(local_file="static/images/banner-ng.svg")
#print(path)

with ui.row().classes('items-center w-full grid gap-4 md:grid-cols-2 bg-fixed bg-cover').style(f"background-image: url('{path}')"):
```
Disabling these doesn't change the 404 problem though.
